### PR TITLE
modification for edged config

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
**What type of PR is this?**

/kind cleanup
/kind feature

**What this PR does / why we need it**:

- modification some parameters in edged config to keep consistent with v1.27.7 kubeletconfig and kubeletflag 
- add DEPRECATED label for some parameter which has been removed in k8s v1.27.7, and remind users we will remove them in KubeEdge v1.17 
- Refactoring ValidateModuleEdged.

### CODE REVIEW

**Comment by ZhengXinwei-F** [comment]
lgtm

File: edge/pkg/edged/edged.go [hunk 1]
```diff
@@ -166,11 +167,19 @@ func newEdged(enable bool, nodeName, namespace string) (*edged, error) {
 	if !edgedconfig.Config.RegisterNode {
 		kubeletConfig.RegisterNode = false
 	}
+
+	// set feature gates from initial flags-based config
+	if err := utilfeature.DefaultMutableFeatureGate.SetFromMap(kubeletConfig.FeatureGates); err != nil {
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:172]: Without the fix above, the FeatureGate will doesn't work, right?

Comment (fisherxu) [reply edge/pkg/edged/edged.go:172]: If so, it should be a bug fix? And should be cherry picked to releases

Comment (Shelley-BaoYue) [reply edge/pkg/edged/edged.go:172]: > Without the fix above, the FeatureGate will doesn't work, right?

Yes, I will separate it to another bugfix pr.

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @WillardHu

**Pull request merged**